### PR TITLE
pythonPackages.requests-hawk: init at 1.0.1

### PR DESCRIPTION
**Pull request by austinbutler**
###### Motivation for this change

Step two of three for fixing the Alerta package for ZHF, as stated in https://github.com/NixOS/nixpkgs/pull/98336. I've tested that the Alerta failure is fixed when this package exists and is added to the Alerta package.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/python-modules/requests-hawk/default.nix [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+    sha256 = "1qcga289yr6qlkmc6fjk0ia6l5cg0galklpdzpslf1y8ky9zb7rl";
+  };
+
+  propagatedBuildInputs = [ mohawk requests ];
+
+  meta = {
+    description = "Hawk authentication strategy for the requests python library.";
+    homepage = "https://github.com/sam-washington/requests-hawk";
+    license = licenses.asl20;
+    maintainers = [ ];
```
Comment (jonringer) [inline pkgs/development/python-modules/requests-hawk/default.nix:19]: please add yourself as maintainer

File: pkgs/development/python-modules/requests-hawk/default.nix [hunk 2]
```diff
@@ -0,0 +1,21 @@
+{ lib, buildPythonPackage, fetchPypi, python, mohawk, requests }:
+
+with lib;
```
Comment (jonringer) [inline pkgs/development/python-modules/requests-hawk/default.nix:3]: only used in meta
```suggestion

```

File: pkgs/development/python-modules/requests-hawk/default.nix [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+  version = "1.0.1";
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "1qcga289yr6qlkmc6fjk0ia6l5cg0galklpdzpslf1y8ky9zb7rl";
+  };
+
+  propagatedBuildInputs = [ mohawk requests ];
+
+  meta = {
```
Comment (jonringer) [inline pkgs/development/python-modules/requests-hawk/default.nix:15]:
```suggestion
  meta = with lib; {
```

File: pkgs/development/python-modules/requests-hawk/default.nix (hunk 1 above)
Comment (austinbutler) [reply pkgs/development/python-modules/requests-hawk/default.nix:19]: Same conversation as last week :smile: : https://github.com/NixOS/nixpkgs/pull/98336#discussion_r491759482.

Comment (jonringer) [reply pkgs/development/python-modules/requests-hawk/default.nix:18]: There's over 3000 python packages in nixpkgs, it's nice to have other have context on a package.

Comment (austinbutler) [reply pkgs/development/python-modules/requests-hawk/default.nix]: Done!

File: maintainers/maintainer-list.nix [hunk 4]
```diff
@@ -806,6 +806,12 @@
     githubId = 574938;
     name = "Jonathan Glines";
   };
+  austinbutler = {
+    email = "austinabutler@gmail.com";
+    github = "austinbutler";
+    githubId = 354741;
+    name = "Austin Butler";
+  };
```
Comment (jonringer) [inline maintainers/maintainer-list.nix:814]: Please make this into a separate commit:

```
maintainers: add <name>
```

File: pkgs/development/python-modules/requests-hawk/default.nix [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+    sha256 = "1qcga289yr6qlkmc6fjk0ia6l5cg0galklpdzpslf1y8ky9zb7rl";
+  };
+
+  propagatedBuildInputs = [ mohawk requests ];
+
+  meta = with lib; {
+    description = "Hawk authentication strategy for the requests python library.";
+    homepage = "https://github.com/sam-washington/requests-hawk";
+    license = licenses.asl20;
+    maintainers = [ austinbutler ];
```
Comment (jonringer) [inline pkgs/development/python-modules/requests-hawk/default.nix:18]:
```suggestion
    maintainers = with maintainers; [ austinbutler ];
```

File: maintainers/maintainer-list.nix (hunk 4 above)
Comment (austinbutler) [reply maintainers/maintainer-list.nix:814]: Done.

**Review by jonringer — APPROVED** [review]
LGTM

```
https://github.com/NixOS/nixpkgs/pull/98881
3 packages built:
python27Packages.requests-hawk python37Packages.requests-hawk python38Packages.requests-hawk
```
